.github/workflows/xlsx-to-csv.yml

```
@@ -15,34 +14,55 @@ jobs:
+        XLSX_CHANGED=$(echo "$FILES" | jq '[.[].filename | select(endswith(".xlsx"))] | length')
+        echo "xlsx_changed=$XLSX_CHANGED" >> $GITHUB_OUTPUT
+
+    - name: Skip if no xlsx changes
+      if: steps.check_xlsx.outputs.xlsx_changed == '0' && github.event_name != 'workflow_dispatch'
+      run: |
+        echo "No xlsx files changed. Skipping workflow."
+        exit 0
+
     - name: Clone the repository
+      if: steps.check_xlsx.outputs.xlsx_changed != '0' || github.event_name == 'workflow_dispatch'
       run: |
```

actionlint (1.7.12)

[error] "github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)